django/db/models/sql/compiler.py

```diff
@@ -1811,23 +1811,54 @@ def as_sql(self):
             on_conflict=self.query.on_conflict,
         )
         result = ["%s %s" % (insert_statement, qn(opts.db_table))]
-        fields = self.query.fields or [opts.pk]
-        result.append("(%s)" % ", ".join(qn(f.column) for f in fields))
 
-        if self.query.fields:
-            value_rows = [
-                [
-                    self.prepare_value(field, self.pre_save_val(field, obj))
-                    for field in fields
+        if fields := self.query.fields:
+            from django.db.models.expressions import DatabaseDefault
+
+            supports_default_keyword_in_bulk_insert = (
+                self.connection.features.supports_default_keyword_in_bulk_insert
+            )
+            value_cols = []
+            for field in list(fields):
+                field_prepare = partial(self.prepare_value, field)
+                field_pre_save = partial(self.pre_save_val, field)
+                field_values = [
+                    field_prepare(field_pre_save(obj)) for obj in self.query.objs
                 ]
-                for obj in self.query.objs
-            ]
+                if field.has_db_default():
+                    # If all values are DEFAULT don't include the field and its
+                    # values in the query as they are redundant and could prevent
+                    # optimizations. This cannot be done if we're dealing with the
+                    # last field as INSERT statements require at least one.
+                    if len(fields) > 1 and all(
+                        isinstance(value, DatabaseDefault) for value in field_values
+                    ):
+                        fields.remove(field)
```

Comment:
Is mutating `self.query.fields` here safe?

Reply:
Good catch. In practice `InsertQuery` is _terminal_ and not chained but in practice it's just bad practice to mutate and object like that.